packages/atd/atd.2.3.2/opam

```diff
@@ -0,0 +1,63 @@
+opam-version: "2.0"
+synopsis: "Parser for the ATD data format description language"
+description: """
+ATD is the OCaml library providing a parser for the ATD language and various
+utilities. ATD stands for Adjustable Type Definitions in reference to its main
+property of supporting annotations that allow a good fit with a variety of data
+formats. """
+maintainer: [
+  "Louis Roché <louis@louisroche.net>"
+  "Martin Jambon <martin@mjambon.com>"
+  "Rudi Grinberg <me@rgrinberg.com>"
+]
+authors: [
+  "Martin Jambon <martin@mjambon.com>"
+  "David Sheets <sheets@alum.mit.edu>"
+  "Rudi Grinberg <me@rgrinberg.com>"
+  "Ivan Jager <aij+git@mrph.org>"
+  "Jeff Meister <nanaki@gmail.com>"
+  "Carmelo Piccione <carmelo.piccione@gmail.com>"
+  "Raman Varabets <roman.vorobets@gmail.com>"
+  "Mathieu Baudet <mathieubaudet@fb.com>"
+  "Rauan Mayemir <rauan@mayemir.io>"
+  "Louis Roché <louis@louisroche.net>"
+  "Brendan Long <self@brendanlong.com>"
+  "Christophe Troestler <christophe.Troestler@umons.ac.be>"
+  "Vincent Bernardoff <vb@luminar.eu.org>"
+  "haoyang <haoyang@esper.co>"
+]
+license: "MIT"
+homepage: "https://github.com/ahrefs/atd"
+bug-reports: "https://github.com/ahrefs/atd/issues"
+depends: [
+  "ocaml" {>= "4.08"}
+  "dune" {>= "2.0"}
+  "menhir"
+  "easy-format"
+  "alcotest" {with-test}
+  "odoc" {with-doc}
+  "re" {>= "1.7.2"}
+]
+dev-repo: "git+https://github.com/ahrefs/atd.git"
+build: [
+  ["dune" "subst"] {pinned}
```

Comment:
```suggestion
  ["dune" "subst"] {dev}
```

Reply:
I don't know what this is. It looks like it's generated by `dune` or `dune-release` from the data found in `dune-project`. Let me figure this out.

Reply:
https://opam.ocaml.org/doc/Manual.html#Package-variables says:

> `pinned`: whether the package is pinned

> `dev`: true if this is a development package, i.e. it was not built from a release archive

I'm fine with whatever works but I don't want to postprocess the output of dune-release if I don't have to. Do you know if it's a bug in dune-release or if I'm using it wrong?

Reply:
Ths is a bug in the opam generation by dune, it is fixed i dune 2.7 or 2.8 but we can fix them here by hand before merging, so don’t worry. I should be able to do a proper review this evening, mentioning some more meaningful changes if needed

Reply:
I'm using dune 2.9.1, which creates or updates the local *.opam files when running `dune build`. These files contain `["dune" "subst"] {pinned}`, not `["dune" "subst"] {dev}`. It looks like it wasn't fixed or that the fix hasn't been released.

Reply:
Yes but your dune file declares the dune Lang version 2.0. They backward compatibility goes all the way down to this bug

Reply:
I just changed to dune lang version 2.8 and it fixes one occurrence of `["dune" "subst"] {pinned}` to `["dune" "subst"] {dev}` (in `atdpy.opam`) but not in the other opam files. I couldn't see why. I'll fix the other occurrences manually.